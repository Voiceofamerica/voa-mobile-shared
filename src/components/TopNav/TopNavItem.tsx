
import * as React from 'react'

import { topNavItem, selectedItem } from './TopNav.scss'

export interface Props extends React.Props<any> {
  selected?: boolean
  onClick?: () => void
}

function TopNavItem (props: Props) {
  const { children, selected, onClick = () => null } = props

  const className = selected
                  ? `${topNavItem} ${selectedItem}`
                  : topNavItem

  return (
    <button className={className} onClick={() => onClick()}>
      {children}
    </button>
  )
}

export default TopNavItem
