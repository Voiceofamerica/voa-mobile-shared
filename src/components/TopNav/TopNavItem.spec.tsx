
import * as React from 'react'
import { create } from 'react-test-renderer'
import { shallow } from 'enzyme'

import TopNavItem from './TopNavItem'

describe('<IconItem />', () => {
  describe('shapshots', () => {
    it('should render empty', () => {
      const element = create((
        <TopNavItem />
      ))

      expect(element).toMatchSnapshot()
    })

    it('should render with the given children', () => {
      const element = create((
        <TopNavItem>
          <div>child1</div>
          <div>child2</div>
        </TopNavItem>
      ))

      expect(element).toMatchSnapshot()
    })

    it('should render selected if prop set to true', () => {
      const element = create((
        <TopNavItem selected />
      ))

      expect(element).toMatchSnapshot()
    })
  })

  describe('onClick', () => {
    it('should be called whenever the element is clicked', () => {
      const onClick = jest.fn()

      const item = shallow((
        <TopNavItem onClick={onClick} />
      ))

      item.simulate('click')

      expect(onClick).toHaveBeenCalledTimes(1)
    })

    it('should not throw an error if not provided and the element is clicked', () => {
      const item = shallow((
        <TopNavItem />
      ))

      item.simulate('click')
    })
  })
})
