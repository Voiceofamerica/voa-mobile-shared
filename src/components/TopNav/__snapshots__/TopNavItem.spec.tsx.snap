// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopNavItem /> shapshots should render empty 1`] = `
<div
  className="topNavItem"
  onClick={[Function]}
  style={undefined}
/>
`;

exports[`<TopNavItem /> shapshots should render selected if prop set to true 1`] = `
<div
  className="topNavItem"
  onClick={[Function]}
  style={undefined}
/>
`;

exports[`<TopNavItem /> shapshots should render with the given children 1`] = `
<div
  className="topNavItem"
  onClick={[Function]}
  style={undefined}
>
  <div>
    child1
  </div>
  <div>
    child2
  </div>
</div>
`;

exports[`<TopNavItem /> shapshots should render with the specified style 1`] = `
<div
  className="topNavItem"
  onClick={[Function]}
  style={
    Object {
      "border": "3px solid #FFFFFF",
    }
  }
/>
`;
